testsuite/source/app.d

```diff
@@ -0,0 +1,323 @@
+            //< means dedent.
+            if (sectionParams.canFind("<"))
+            {
+                sectionData = sectionData[4..$].substitute("\n    ", "\n", "<SPC>", " ", "<TAB>", "\t").toUTF8;
+            }
+            else
+            {
+                sectionData = sectionData.substitute("<SPC>", " ", "<TAB>", "\t").toUTF8;
+            }
+            //Not sure what + means.
+        }
+        switch(sectionName)
+        {
+            case "in-yaml":
+                parseYAML(sectionData);
+                break;
+            case "in-json":
+                json = parseJSON(sectionData);
+                break;
+            case "test-event":
+                events = sectionData;
+                break;
+            case "error":
+                shouldFail = true;
+                testsRun++;
+                break;
+            case "out-yaml":
+                compareYAMLString = sectionData;
+                break;
+            case "emit-yaml":
+                // TODO: Figure out how/if to implement this
+                //fail("Unhandled test - emit-yaml");
+                break;
+            case "lex-token":
+                // TODO: Should this be implemented?
+                //fail("Unhandled test - lex-token");
+                break;
+            case "from": break;
+            case "tags": break;
+            default: assert(false, text("Unhandled section ", sectionName, "in ", output.name));
+        }
+    }
+    if (!loadFailed && !compareYAMLString.isNull && !shouldFail)
+    {
+        Appender!string buf;
+        dumper(buf).dump();
+        compareLineByLine(buf.data, compareYAMLString, "Dumped YAML mismatch");
+        testsRun++;
+    }
+    if (!loadFailed && !events.isNull && !shouldFail)
+    {
+        const compare = dumpEventString(yamlString);
+        compareLineByLine(compare, events, "Event mismatch");
+        testsRun++;
+    }
+    if (loadFailed && !shouldFail)
+    {
+        fail(failMsg);
+    }
+    if (shouldFail && !loadFailed)
+    {
+        fail("Invalid YAML accepted");
+    }
+    if ((testsRun == 0) && (output.state != TestState.failure))
+    {
+        skip("No tests run");
+    }
+    return output;
+}
+
+// Can't be @safe due to dirEntries()
+void main(string[] args) @system
+{
+    string path = "yaml-test-suite/test";
+
+    void printResult(string id, TestResult result)
+    {
+        writeln(id, " ", result);
+    }
+
+    if (args.length > 1)
+    {
+        path = args[1];
+    }
+
+    ulong total;
+    ulong successes;
+    foreach (file; dirEntries(path, "*.tml", SpanMode.shallow))
+    {
+        auto result = runTests(readText(file));
+        if (result.state == TestState.success)
+        {
+            debug(verbose) printResult(file.baseName, result);
+            successes++;
+        }
+        else
+        {
+            printResult(file.baseName, result);
+        }
+        total++;
+    }
+    writefln!"%d/%d tests passed"(successes, total);
```

Comment:
I dont see this log line on TravisCI log

Reply:
because travis-CI doesn't run the binary. we don't pass all the tests yet.

Reply:
it would mean every PR is blocked until we fully implement YAML 1.2, which is pretty far away